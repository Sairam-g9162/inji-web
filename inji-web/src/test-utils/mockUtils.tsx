import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { reduxStore } from '../redux/reduxStore';

export const mockUseTranslation = () => {
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
            t: (key: string) => key,
        }),
    }));
};

export const mockUseNavigate = () => {
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useNavigate: jest.fn(),
    }));
};

export const wrapUnderRouter = (children: React.ReactNode) => {
    return <Router>{children}</Router>;
};

interface RenderWithProviderOptions extends Omit<RenderOptions, 'queries'> {}

const renderWithProvider = (ui: ReactElement, options?: RenderWithProviderOptions) => {
    return render(
        <Provider store={reduxStore}>
            <Router>
                {ui}
            </Router>
        </Provider>,
        options
    );
};

export * from '@testing-library/react';
export { renderWithProvider };
