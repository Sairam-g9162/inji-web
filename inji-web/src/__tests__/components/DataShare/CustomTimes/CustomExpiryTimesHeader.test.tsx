import {screen} from "@testing-library/react";
import {CustomExpiryTimesHeader} from "../../../../components/DataShare/CustomExpiryTimes/CustomExpiryTimesHeader";
import { renderWithProvider } from "../../../../test-utils/mockUtils";


describe("Test the Layout of the Custom Expiry Header", () => {
    test("Test the presence of the Outer Container", ()=>{
        const{asFragment} = renderWithProvider(<CustomExpiryTimesHeader title={"CTHeader"} />)
        expect(asFragment()).toMatchSnapshot();
    })
   
});
describe("Test the functionality of custom expiry Header",()=>{
    test("Test to Have the content", ()=>{
        renderWithProvider(<CustomExpiryTimesHeader title={"CTHeader"} />);
        const document = screen.getByTestId("CustomExpiryTimesHeader-Title-Content");
        expect(document).toHaveTextContent("CTHeader");
    })
});
