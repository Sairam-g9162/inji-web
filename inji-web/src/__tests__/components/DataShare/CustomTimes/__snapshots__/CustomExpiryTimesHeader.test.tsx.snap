// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Layout of the Custom Expiry Header Test the presence of the Outer Container 1`] = `
<DocumentFragment>
  <div
    class="flex items-start justify-between p-5 rounded-t text-center"
    data-testid="CustomExpiryTimesHeader-Outer-Container"
  >
    <div
      class="text-lg font-semibold"
      data-testid="CustomExpiryTimesHeader-Title-Content"
    >
      CTHeader 
    </div>
  </div>
</DocumentFragment>
`;
